include/openthread/ip6.h

```diff
@@ -97,6 +97,21 @@ struct otIp6Prefix
  */
 typedef struct otIp6Prefix otIp6Prefix;
 
+/**
+ * IPv6 Address types
+ *
+ */
+enum
+{
+    kLinkLocal = 0, ///< Link-local address
+    kSlaac     = 1, ///< SLAAC assigned address
+    kDhcpv6    = 2, ///< DHCPv6 assigned address
+    kManual    = 3, ///< Manually assigned address
```

Comment:
I don't think these are all valid address types. 

SLAAC, DHCP, MANUAL are actually types of methods for generating addresses. 
LinkLocal, Aloc, MeshLocal, DUA are address types, but they can be generally determined by checking network prefixes or interface ids.

Reply:
How about calling the variable "mAddressSource", "mAddressMechnism", or "mAddressMethod" instead? And then have the values kLink, kSlaac, kDhcpv6, kManual, and possibly kMesh?

Reply:
How about using `mAddressOrigin`?

Reply:
I think we should use proper checking for prefixes and locators to determine the address types. 
For the address origin, what about just defining four origin types (using only 2 bits):

```c++
    kSystem = 0; // System generated addresses including ALOCs, MLEID, RLOC, etc
    kSlaac = 1; // SLAAC
    kDhcpv6 = 2; // DHCPv6
    kManual = 3; // Manually configured
```

Thoughts? @markus-becker-tridonic-com @abtink

Reply:
> I think we should use proper checking for prefixes and locators to determine the address types.

Seemed pretty obvious which type/origin it is in the code. When in dhcp6_server.hpp, it is a DHCP address type/origin, etc.